### ISSUE TITLE
Transposed Convolution output shape does not take into account dilation

### ISSUE THREAD

**Issue by Coolnesss** [pytorch/pytorch#14099]
## 📚 Documentation

Hey, there seems to be a simple mistake in the equation for the output of all the transposed convolution layers (e.g https://pytorch.org/docs/master/nn.html#convtranspose2d). The equations do not take into account dilation. 

I believe the correct formula should be (in the 1d case)

```
(L_in - 1) * s - 2 * p + (k - 1)*d + o_p + 1
```

where `L_in, s, p, k, d, o_p` are input length, stride, padding, kernel size, dilation and output padding, respectively.

**Comment by Erotemic** [comment]
I just noticed this as well because I use the formula provided in by the docs in my [OutputShapeFor](https://github.com/Erotemic/netharn/blob/master/netharn/output_shape_for.py) class.

Considering the code

```python.
        module = nn.ConvTranspose2d(1, 1, kernel_size=3, stride=1, dilation=2)
        input_shape = (1, 1, 10, 10)
        module(torch.rand(*input_shape)).shape
```

According  to the docs the output shape should be (1, 1, 12, 12), but in reality the output shape is (1, 1, 14, 14).

Modifying the formula as suggested by @Coolnesss does result in the correct answer, and the changes matches my intuition of how dilation increases the effective kernel size.

**Comment by soumith** [comment]
would either of you be kind enough to send a PR to fix the docstring
